src/nix.rs

```diff
@@ -406,17 +434,34 @@ mod tests {
         nix.argstr("foo", "bar");
 
         let exp: Vec<&OsStr> = [
+            "-A",
+            "hello",
+            "--argstr",
+            "foo",
+            "bar",
             "--expr",
             "my-cool-expression",
+        ]
+        .into_iter()
+        .map(OsStr::new)
```

Comment:
These are already OsStr's though, why do we need to map them?

Reply:
They’re `str`s unfortunately, rust does not change the type of a string literal depending on context.

Reply:
I don't follow. The type hasn't changed, it has always been `let exp: Vec<&OsStr> = [ `, so what changed when adding a handful more arguments to require this new code?

Reply:
`vec!["a", "b", "c"]` is a `Vec<str>`, only with `.map(OsStr::new)` does it become a `Vec<&OsStr>` (the type is for the final result of the full iterator chain).